operator/pkg/client/client_factory.go

```diff
@@ -103,11 +104,24 @@ func NewSingletonClients(config *rest.Config, logger logr.Logger) (*SingletonCli
 	discoveryRESTMapper := restmapper.NewDeferredDiscoveryRESTMapper(cachedDiscoveryClient)
 	discoveryShortcutExpander := restmapper.NewShortcutExpander(discoveryRESTMapper, cachedDiscoveryClient)
 
-	kubernetesClient, err := kubernetes.NewForConfigAndClient(config, httpClient)
+	// Create target cluster client only if not passed.
+	// Clients should be passed only in two cases:
+	// 1. Single cluster mode is enabled.
+	// Since such clients are similar to the root client instance.
+	// 2. Client instance is explicitly passed from the library interface
+	runtimeClient := info.Client
```

Comment:
could you explain here in which scenario the new client proxy is created?

Reply:
All cases except the two mentioned above lead to a client proxy creation. I can also add this as a comment, but I don't think it's required.